### PR TITLE
ARTEMIS-811 Broker sends unsolicited Detach during connection shutdown

### PR DESCRIPTION
**Pull request by franz1981**

### CODE REVIEW

File: tests/artemis-test-support/src/main/java/org/apache/activemq/transport/amqp/client/AmqpClient.java [hunk 1]
```diff
@@ -76,6 +77,23 @@ public AmqpConnection connect() throws Exception {
...
 
    /**
+    * Creates a connection with the broker at the given location, this method initiates a
+    * connect attempt immediately and will fail if the remote peer cannot be reached.
+    *
+    * @param  protocolTracer handler of all the received and sent frames
+    * @throws Exception if an error occurs attempting to connect to the Broker.
+    * @return a new connection object used to interact with the connected peer.
+    */
+   public AmqpConnection connect(final ProtocolTracer protocolTracer) throws Exception {
```
Comment (clebertsuconic) [inline tests/artemis-test-support/src/main/java/org/apache/activemq/transport/amqp/client/AmqpClient.java]: @tabish121  wouldn't this needed to by sync with AMQ5 code?

Comment (tabish121) [inline tests/artemis-test-support/src/main/java/org/apache/activemq/transport/amqp/client/AmqpClient.java]: Not sure I'd do it this way so I'd prefer not to port this to 5.x, the client already has some support for internal inspectors which probably could be modified to do this without needing this new interface and massive stub in the test

**Comment by clebertsuconic** [comment]
@franz1981  I don't think this is ready to be merged. I don't see a fix anyways?

**Comment by franz1981** [comment]
@clebertsuconic The test doesn't fail against the current release but is it not good to have a test to be sure that bug will never happen in the future?

File: tests/artemis-test-support/src/main/java/org/apache/activemq/transport/amqp/client/AmqpClient.java (hunk 1 above)
Comment (franz1981) [inline tests/artemis-test-support/src/main/java/org/apache/activemq/transport/amqp/client/AmqpClient.java]: @tabish121 what is the way you suggest to intercept frames received from the client?I've searched for something similar and I've found only a state validator,but it is not the same.Maybe I've missed something?

**Pull request closed without merging**